src/marten/db/query/set.cr

```diff
@@ -325,6 +325,40 @@ module Marten
           add_query_node(query_node)
         end
 
+        # Returns if a query set includes a specific model.
+        def includes?(value : DB::Model)
+          query_node = Node.new({Constants::PRIMARY_KEY_ALIAS => value.pk})
+          add_query_node(query_node)
+          exists?
+        end
+
+        # Returns if a query set matches a specific set of advanced filters.
+        #
+        # This method returns a `Bool` object and allows to define complex database queries involving
+        # **AND** and **OR** operators. It yields a block where each filter has to be wrapped using a `q(...)`
+        # expression. These expressions can then be used to build complex queries such as:
+        #
+        # ```
+        # query_set = Post.all
+        # query_set.includes? { (q(name: "Foo") | q(name: "Bar")) & q(is_published: true) }
+        # ```
+        def includes?(&)
+          expr = Expression::Filter.new
+          query : Node = with expr yield
+          filter(query)
+          exists?
+        end
+
+        # Returns if a query set includes a specific node.
+        # ```
+        # query_set = Post.all
+        # query_set.includes?(Marten::DB::Query::Node.new(name__startswith: "Fr"))
+        # ```
+        def includes?(query_node : Node)
+          filter(query_node)
+          exists?
+        end
```

Comment:
I am not opposed to adding these methods. Though in my mind they should be called `#exists?` since what we are doing is verifying that the query set produces results (which would also be more consistent with the existing `#exists?` method). Can we rename those to `#exists?`?

Reply:
You can simply rename these methods to `#exists?` because afaik `#includes?` has only [one interface](https://crystal-lang.org/api/1.7.2/Enumerable.html#includes%3F%28obj%29%3ABool-instance-method) where you specify one object as argument (ie. the object to use for the inclusion check).